### DISCUSSION TITLE
[Feature] Allow option for blurred thumbnails, and album covers for private photos.

### DISCUSSION THREAD

**Discussion by relink2013** [immich-app/immich#12256] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

There already seems to be some fairly active discussion on similar topic in #2451, however that feature discussion is way more complex than what Im looking for. 

All I'd like to do is be able to set certain pictures/videos and albums as "sensitive". It could be a check box, a specific tag, whatever is easiest to implement to be honest. 

All that would happen with an image or video thats marked as sensitive is the thumbnail would be blurred. Same if an album were marked as sensitive, just blur the album cover. 

When you tap the thumbnail, the original non-blurred version is loaded. 

A toggle in the UI to blur/un-blur all would be nice though. This way you don't need to tap every single image to find the one you're looking for.

Thats literally it, no passwords, no hiding from the timeline, just a blurred thumbnail with a toggle. 

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by CyberCreator** [comment]
I support your idea — it's exactly what I’ve been looking for. Why? Because it would allow us to easily blur out certain sensitive photos. For example, imagine you have pictures from a surgery you went through — you probably don’t want those popping up when casually scrolling through your timeline with others around, just trying to find a specific photo from a specific date. It would be awkward if such “special photos” suddenly appeared. A tag that triggers a blur would be perfect.

**Comment by bo0tzz** [comment]
#2451 is now implemented, superseding this request.

**Reply by relink2013** [reply]
It’s been a while since I’ve used Immich but now I really am trying to make Immich my new primary. 

I read through [#2451](https://github.com/immich-app/immich/discussions/2451) as well as a few other FR on this topic and they absolutely do not fill the need I was originally hoping for. In fact most of them are—no offense—clunky or wilding over complicated. 

What I was originally trying to purpose was simple, just create a blurred thumbnail for select images and provide a toggle to temporarily show the non blurred thumbnail when needed. 

For better implementation and a smooth experience I thought of 3 scenarios;

1. Image is tagged “nsfw” the timeline thumbnail gets blurred. Optionally the preview can be blurred too with a simple temporary “show image” toggle when viewing. 

2. If an entire album is marked as “nsfw” then any images in that album automatically get a blurred thumbnail in the main timeline, but are shown un-blurred if that album is opened directly and only in that album. 

3. If there are nsfw images in a regular album, then treat them like the timeline photos. Blur the thumbnail (and optionally the preview), with a simple temporary toggle. 

——

I mean no disrespect to anyone else, but that previous FR got kind of off the rails. Are people storing nuclear launch codes in their Immich library? 

- I dont need password protected albums
- I don’t need hidden albums
- I don’t want the images removed or hidden from my main timeline.
- I don’t want to dump everything in a special hidden album destroying organization and continuity. 

All I want is to be able to blur the images unless I want to view them. I don’t need Fort Knox, just the ability to not accidentally show boobs to the wrong person.

**Reply by leonnicklas** [reply]
@relink2013 This is exactly what I'm looking for too! 
I fully support your idea and am hoping to see this feature very soon on the Immich app as well. 
Other software providers are implementing this feature already (e.g. the native UGREEN NAS photos app!) and it is great!
Locked folders just aren't the right fit for that because I cannot find the pictures anymore when all of them are thrown into one large folder.
I want them in my timeline, in my albums, and in my search, but I just want a toggle to blur/unblur.

@bo0tzz
